pandas/tests/arrays/categorical/test_repr.py

```diff
@@ -253,6 +257,19 @@ def test_categorical_repr_int_with_nan(self):
 Categories (2, int64): [1, 2]"""
         assert repr(s) == s_exp
 
+    @pytest.mark.parametrize("values_dtype", tm.ALL_INT_EA_DTYPES)
+    def test_categorical_repr_nullable_int_NA(self, values_dtype):
+        arr = array([1, 2, np.nan], dtype=values_dtype)
+        c = Categorical(arr)
+        c_exp = f"""[1, 2, {NA}]\nCategories (2, {values_dtype}): [1, 2]"""
```

Comment:
so if you did `c[2]` would you get pd.NA or np.nan?  i'm wary of changing the repr without changing the item